clang-tools-extra/clang-tidy/utils/FixItHintUtils.cpp

```diff
@@ -71,15 +72,20 @@ static std::optional<FixItHint> fixIfNotDangerous(SourceLocation Loc,
 
 // Build a string that can be emitted as FixIt with either a space in before
 // or after the qualifier, either ' const' or 'const '.
-static std::string buildQualifier(DeclSpec::TQ Qualifier,
+static std::string buildQualifier(Qualifiers::TQ Qualifier,
                                   bool WhitespaceBefore = false) {
   if (WhitespaceBefore)
-    return (llvm::Twine(' ') + DeclSpec::getSpecifierName(Qualifier)).str();
-  return (llvm::Twine(DeclSpec::getSpecifierName(Qualifier)) + " ").str();
+    return (llvm::Twine(' ') +
+            DeclSpec::getSpecifierName(static_cast<DeclSpec::TQ>(Qualifier)))
```

Comment:
I feel like this is moving the problem to the source file. Can we use `Qualifiers::getAsString()` and remove the `DeclSpec` dependence entirely?

Reply:
> Qualifiers::getAsString()

From that I saw, it does much more than `getSpecifierName`. It also can serialize only a subset of qualifiers, because `Qualifiers::TQ` doesn't seem to support `__unaligned` and `_Atomic`. I think what I've done is an improvement over status quo, and I'd like to avoid going into the rabbit hole.

Reply:
In practice, this function only ever to print "const", afaik (and nothing else).

You could even reimplement it inline

```cpp
const char getSpecifierName(Qualifier T) {
  switch (T) {
     case Qualifiers::Qualifier:   return "const";
     default:
          assert("not supported");
  }
  return "";
}
```

But really if `Qualifiers::getAsString` passes the test, you should just use that.

Reply:
I updated the PR. Is it good now?

Reply:
LGTM.